### ISSUE TITLE
New migration doesn't run on tenant

### ISSUE THREAD

**Issue by bkintanar** [hyn/multi-tenant#104]
Ok here's the gist of my situation. I created a set of new migrations and upon issuing `php artisan migrate --tenant=all` it doesn't migrate the tenant database with the new migrations. Please see steps below.

PS. This only works if you apply my PR: https://github.com/hyn/multi-tenant/pull/103
1. blank master database
2. no tenant databases yet
3. I have migrations after hyn/multi-tenant migrations.
4. I then run `multi-tenant:setup` as seen below to create new tenant. New migrations as highlighted in the screenshot shows at the bottom. No surprises there since that's how it should work.
   <img width="1158" alt="screen shot 2016-06-11 at 9 36 42 am" src="https://cloud.githubusercontent.com/assets/685928/15982455/8e77da84-2fba-11e6-957d-f23f30ff6bb3.png">
5. I then ran migrations for tenant `php artisan migrate --tenant=all`
6. As you can see on the screenshot the 3 migrations from above doesn't appear in the tenant migration.
   <img width="570" alt="screen shot 2016-06-11 at 9 36 59 am" src="https://cloud.githubusercontent.com/assets/685928/15982456/8e7a5e26-2fba-11e6-930e-8d164ceacb9e.png">
7. I reset the master database, deleted all the tables, removed the newly created tenant.
8. I then tested it out and renamed all 3 migrations in such a way that it will show up first on the list and that the hyn/multi-tenant migrations will be shown last. As you can see on the screenshot it works as expected on the master database, the 3 migrations are now shown first.
   <img width="1154" alt="screen shot 2016-06-11 at 9 42 40 am" src="https://cloud.githubusercontent.com/assets/685928/15982457/8e7edf0a-2fba-11e6-913f-ad0f30ddc941.png">
9. I then tried again and run the migrations on the tenant created. This time the 3 migrations hows on top of the hyn/multi-tenant migrations as expected.
   <img width="545" alt="screen shot 2016-06-11 at 9 43 08 am" src="https://cloud.githubusercontent.com/assets/685928/15982458/8e8752f2-2fba-11e6-9b4f-6a7103fbaae6.png">

**Comment by luceos** [comment]
Didn't I do the same here?

https://github.com/hyn/multi-tenant/pull/98/files

**Comment by bkintanar** [comment]
@Luceos I'm afraid not. There are two hostnames_table around. hmt_hostnames ad hws_ssl_hostnames and you missed the following files below.

```
hws_ssl_certificates_table.php
hws_ssl_hostnames_table.php
```

**Comment by luceos** [comment]
i see great. The diff is really unreadable

**Comment by luceos** [comment]
closed in https://github.com/hyn/multi-tenant/pull/103

**Comment by bkintanar** [comment]
Why close? My PR is needed for this to be tested but this still is occurring.

**Comment by luceos** [comment]
I merged your PR, thinking that solved this issue. Did I miss something?

**Comment by bkintanar** [comment]
Yes, the PR is needed for the tenancy migrations to work. but even if we merged my PR, any new migrations that are added after tenancy's migration doesn't run.

**Comment by bkintanar** [comment]
Should this issue be reopened?

**Comment by luceos** [comment]
@bkintanar are you able to verify the recent changes fixed this?

I tagged 2.0.0-beta.1

**Comment by bkintanar** [comment]
@Luceos let me check and see. I'll get back to you with update.

**Comment by bkintanar** [comment]
@Luceos this issue is still occurring.

**Comment by luceos** [comment]
@bkintanar are you using the 5.2 compatible one? Might be the case that native Laravel command was modified and the package doesn't reflect those changes.

**Comment by bkintanar** [comment]
@Luceos I'm not sure I understand. I just created a normal migration file using command and then I tried creating manually. Is this your question?

If you're question if I'm using the 5.2 compatible branch then yes.

**Comment by MithrandirDK** [comment]
I am using Laravel 5.3 and Hyn 2.0\* and new migrations are not run on my tenant databases.

I don't get any errors about failed tables etc.

Once I rename the migration to 2014_something, it runs perfectly on the tenants.

**Comment by bkintanar** [comment]
@MithrandirDK this is still an open issue.

**Comment by MithrandirDK** [comment]
Right, I just hoped to shed some light on the issue and ease the reproducing - since I experience it on a very clean Laravel 5.3 / Hyn 2.0\* installation - not any 5.2 compatible branch or anything.

It was not meant to be a rant or otherwise criticism of this really excellent component that will prove to be essential in my current project :)

**Comment by deleugpn** [comment]
I just reproduced this bug on a clean install of Laravel 5.3 and Hyn 2.0.x@dev. All I did was `php artisan make:migration` to create a new migration and when running `php artisan migrate --tenant=1` I had the following scenario: 

```
[root@php7 hello-world]# php artisan migrate --tenant=1
Migrating for 1: home-php7-
Migration table created successfully.
Migrated: 2014_10_12_000000_create_users_table
Migrated: 2014_10_12_100000_create_password_resets_table
```

When I ran it again, I got this: 

```
[root@php7 hello-world]# php artisan migrate --tenant=1
Migrating for 1: home-php7-
Migrated: 2015_05_01_111209_hmt_tenants_table
Migrated: 2015_05_01_111210_hmt_websites_table
Migrated: 2015_05_01_111211_hmt_hostnames_table
Migrated: 2015_05_19_000000_hws_ssl_certificates_table
Migrated: 2015_05_19_000001_hws_ssl_hostnames_table
```

but none of these tables were actually created in the tenant database. However, his `migrations` table registered those values.

Then I ran it again: 

```
[root@php7 hello-world]# php artisan migrate --tenant=1
Migrating for 1: home-php7-
```

And nothing would happen. After finding this topic and following @MithrandirDK's advice (renaming my migration to `2014_x`, I successfully ran it: 

```
[root@php7 hello-world]# php artisan migrate --tenant=1
Migrating for 1: home-php7-
Migrated: 2014_09_18_190257_create_posts_table
[root@php7 hello-world]#
```

Since this is a known issue, are there any suggestions to overcome this problem aside from renaming the migrations to `2014_`? Is it okay to do this for now?

**Comment by bkintanar** [comment]
@Deleugpn yes. currently my workaround above is the way to do this. The team is working so that this will be fixed before 2.0.0 release. Please stay tuned and sorry for this inconvenience.

**Comment by MithrandirDK** [comment]
For improved debugging, I find that we need a change in MigrateCommand::fire() as it swallows any PDOExceptions and only give output when the error contains a very specific string.

```
try {
    $this->migrator->run($path, [
        'pretend' => $pretend,
        'step' => $this->input->getOption('step'),
    ]);
    } catch (PDOException $e) {
        if (str_contains($e->getMessage(), ['Base table or view already exists'])) {
            $this->comment("Migration failed for existing table; probably a system migration: {$e->getMessage()}");
            continue;
        }
   }
```

I would suggest a 

```
else {
   throw $e;
}
```

inside the catch() block

**Comment by MithrandirDK** [comment]
This change was included in #174 , so are you now getting more information out of a migrate run?

**Comment by luceos** [comment]
Obsoleted with the work on 3.x
